New Page: aria-level (#10439)
mdn/content: files/en-us/web/accessibility/aria/attributes/aria-level/index.md

---
title: 'aria-level'
slug: Web/Accessibility/ARIA/Attributes/aria-level
tags: 
  - Accessibility
  - ARIA
  - ARIA attribute
  - ARIA property
  - aria-level
  - Reference
---

The `aria-level` attribute defines the hierarchical level of an element within a structure.

## Description

> When it comes to the headings in a document structure, you can have primary headings, secondary headings, tertiary headings, and so on. In trees, you have the root element, its children, the children's children (or grandchildren), and so on. Levels of hierarchy appear in headings, trees, nested grid, nested tablists, and more. If the DOM ancestry does not accurately represent the level, the `aria-level` attribute should be used to define the hierarchical level elements within their hierarchical structures. Levels increase with depth. The value for aria-level is an integer greater than or equal to 1.

Reviewer: ```suggestion
Levels of hierarchy appear in headings, trees, nested grid, nested tablists, and more. If the DOM ancestry does not accurately represent the level, the `aria-level` attribute should be used to define the hierarchical level elements within their hierarchical structures. Levels increase with depth. The value for aria-level is an integer greater than or equal to `1`.

When it comes to the headings in a document structure, you can have first-level headings, second-level headings, third-level headings, etc. In trees, you have the root element, its children, the children's children (or grandchildren), and so on. 
```

> The `aria-level` attribute exposes the hierarchy to assistive technologies so the hierarchy can be communicated to the users. Like all ARIA attributes, it has no impact on the user agent and therefore no impact on the user agent's determination of the document structure.

Reviewer: ```suggestion
The `aria-level` attribute exposes hierarchy to assistive technologies so it can be communicated to the users. Like all ARIA attributes, it has no impact on the user agent and therefore no impact on the user agent's determination of the document structure.
```

> If the DOM ancestry accurately represents the level, the user agent can calculate the level of an item from the document structure making `aria-level` not only redundant but a risk of creating misinformation. This attribute should really only be used to provide an explicit indication of the level when that is not possible to calculate from the document structure. Test to see if this attribute is needed. If the user agent can calculate the level, it is best to omit the `aria-level` attribute.

Reviewer: ```suggestion
If the DOM ancestry accurately represents the level, the user agent can calculate the level of an item from the document structure, making `aria-level` not only redundant but a risk of creating misinformation. `aria-level` should really only be used to provide an explicit indication of level when it is not possible to calculate from the document structure. Test to see if this attribute is needed. If the user agent can calculate the level, it is best to omit the `aria-level` attribute.
```
